wallet/src/main/scala/org/bitcoins/wallet/models/AddressTable.scala

```diff
@@ -0,0 +1,147 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.config.NetworkParameters
+import org.bitcoins.core.crypto.bip44.{BIP44ChainType, BIP44Coin, BIP44Path}
+import org.bitcoins.core.crypto.{
+  ECPublicKey,
+  ExtPrivateKey,
+  Sha256Hash160Digest
+}
+import org.bitcoins.core.protocol.script.{
+  P2WPKHWitnessSPKV0,
+  P2WPKHWitnessV0,
+  ScriptWitness
+}
+import org.bitcoins.core.protocol.{Bech32Address, BitcoinAddress}
+import org.bitcoins.core.script.ScriptType
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+
+// todo: make ADT for different addresses in DB, seeing as they have different fields
+// todo: indicate whether or not address has been spent to
+case class AddressDb(
+    path: BIP44Path,
+    ecPublicKey: ECPublicKey,
+    hashedPubKey: Sha256Hash160Digest,
+    address: BitcoinAddress,
+    witnessScriptOpt: Option[ScriptWitness],
+    scriptType: ScriptType)
+
+object AddressDbHelper {
+
+  /** Get a Segwit pay-to-pubkeyhash address */
+  def getP2WPKHAddress(
+      xpriv: ExtPrivateKey,
+      path: BIP44Path,
+      np: NetworkParameters): AddressDb = {
+
+    val xprivAtPath: ExtPrivateKey = xpriv.deriveChildPrivKey(path)
+    val pub = xprivAtPath.key.publicKey
+    val witnessSpk = P2WPKHWitnessSPKV0(pub)
+    val scriptWitness = P2WPKHWitnessV0(pub)
+    val addr = Bech32Address(witnessSpk, np)
+    AddressDb(path,
+              pub,
+              witnessSpk.pubKeyHash,
+              addr,
+              Some(scriptWitness),
+              ScriptType.WITNESS_V0_KEYHASH)
+  }
+}
+
+class AddressTable(tag: Tag) extends Table[AddressDb](tag, "addresses") {
+  import org.bitcoins.db.DbCommonsColumnMappers._
+
+  def accountIndex: Rep[Int] = column[Int]("account_index")
+
+  def accountCoin: Rep[BIP44Coin] = column[BIP44Coin]("bip44_coin")
+
+  def accountChainType: Rep[BIP44ChainType] =
+    column[BIP44ChainType]("bip44_chain_type")
+
+  def addressIndex: Rep[Int] = column[Int]("address_index")
+
+  def address: Rep[BitcoinAddress] =
+    column[BitcoinAddress]("address", O.PrimaryKey)
+
+  def ecPublicKey: Rep[ECPublicKey] = column[ECPublicKey]("pubkey")
```

Comment:
There isn't a one to one relationship between public keys and addresses, for instance multisig. This design needs work